pkgs/servers/x11/xorg/xwayland.nix

```diff
@@ -1,37 +1,31 @@
 { lib, wayland, wayland-protocols, xorgserver, xkbcomp, xkeyboard_config
-, epoxy, libxslt, libunwind, makeWrapper, egl-wayland
-, defaultFontPath ? "" }:
-
-with lib;
+, epoxy, libtirpc, libxslt, libunwind, makeWrapper, egl-wayland, fetchurl, meson, ninja
+, pkg-config, fontutil, defaultFontPath ? "" }:
 
 xorgserver.overrideAttrs (oldAttrs: {
 
-  name = "xwayland-${xorgserver.version}";
-  buildInputs = oldAttrs.buildInputs ++ [ egl-wayland ];
+  name = "xwayland-21.1.0";
+  src = fetchurl {
+    url = "https://xorg.freedesktop.org/archive/individual/xserver/xwayland-21.1.0.tar.xz";
+    sha256 = "sha256-inHD6LlcQ7uRvD16T/kHVFagz+KXch28jUp2zyQdgvs=";
+  };
+  patches = [ ];
+  outputs = [ "out" ];
+  nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ meson ninja ];
+  buildInputs = oldAttrs.buildInputs ++ [ egl-wayland fontutil libtirpc ];
   propagatedBuildInputs = oldAttrs.propagatedBuildInputs
     ++ [wayland wayland-protocols epoxy libxslt makeWrapper libunwind];
-  configureFlags = [
-    "--disable-docs"
-    "--disable-devel-docs"
-    "--enable-xwayland"
-    "--enable-xwayland-eglstream"
-    "--disable-xorg"
-    "--disable-xvfb"
-    "--disable-xnest"
-    "--disable-xquartz"
-    "--disable-xwin"
-    "--enable-glamor"
-    "--with-default-font-path=${defaultFontPath}"
-    "--with-xkb-bin-directory=${xkbcomp}/bin"
-    "--with-xkb-path=${xkeyboard_config}/etc/X11/xkb"
-    "--with-xkb-output=$(out)/share/X11/xkb/compiled"
+  mesonFlags = [
+    "-Dxwayland-eglstream=true"
+    "-Ddefault-font-path=${defaultFontPath}"
+    "-Dxkb_bin_dir=${xkbcomp}/bin"
+    "-Dxkb_dir=${xkeyboard_config}/etc/X11/xkb"
+    "-Dxkb_output_dir=$(out)/share/X11/xkb/compiled"
   ];
 
-  postInstall = ''
-    rm -fr $out/share/X11/xkb/compiled
-  '';
+  postInstall = [ ];
```

Comment:
```suggestion
```

Reply:
Required to remove the patches from the overlay.